x/nexus/keeper/msg_dispatcher.go

```diff
@@ -60,12 +60,15 @@ func (m Messenger) routeMsg(ctx sdk.Context, msg exported.WasmMessage) error {
 		return fmt.Errorf("recipient chain %s is not a registered chain", msg.DestinationChain)
 	}
 
-	id, _, _ := m.GenerateMessageID(ctx)
+	if _, ok := m.Nexus.GetMessage(ctx, msg.ID); ok {
+		return nil
+	}
+
 	sourceChain := exported.Chain{Name: msg.SourceChain, SupportsForeignAssets: false, KeyType: tss.None, Module: wasmtypes.ModuleName}
 	sender := exported.CrossChainAddress{Chain: sourceChain, Address: msg.SourceAddress}
 	recipient := exported.CrossChainAddress{Chain: destinationChain, Address: msg.DestinationAddress}
 
-	nexusMsg := exported.NewGeneralMessage(id, sender, recipient, msg.PayloadHash, msg.SourceTxID, msg.SourceTxIndex, nil)
+	nexusMsg := exported.NewGeneralMessage(msg.ID, sender, recipient, msg.PayloadHash, msg.SourceTxID, msg.SourceTxIndex, nil)
```

Comment:
Let's add `sourceChain` to `msg.ID` as a prefix (`{sourceChain}_{ID}`) for domain separation? Accidental message ID collisions (or a result of compromise) with Amplifier chains could be a big issue otherwise. We also need it if we want to reuse the same AxelarGateway via the consensus and an Amplifier based light client connection (as the event id will be the same)

This prefixing will affect axelarons and axelarscan since they expect it to be just the tx hash - event index rn

Reply:
I don't think it makes sense to prefix. The whole idea of this PR was to just pass message ids between amplifier and core unchanged, for better tracking. If we prefix here, then the message id changes when the message passes from amplifier to core. Also, do we then need to strip the source chain prefix off when going from core to amplifier? I think this would make things very confusing and hard to reason about. We don't prefix with source chain in amplifier, and we aren't prefixing with source chain anywhere in core at the moment, so I don't think we should prefix here.

Reply:
actually, I think it's fine to prefix with the source chain. Just need to communicate with axelarons and any other consumers